plugins/lsp/schemastore.nix

```diff
@@ -0,0 +1,140 @@
+          If a schema with the given name isn't found, the custom schema will not be returned.
+        '';
+
+        extra = helpers.defaultNullOpts.mkListOf schemaEntry "[]" ''
+          Additional schemas to include.
+        '';
+      };
+    in {
+      json = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = "Whether to enable the json schemas in jsonls.";
+        };
+
+        settings = helpers.mkSettingsOption {
+          options = schemaOpts;
+          description = "Options supplied to the `require('schemastore').json.schemas` function.";
+          example = {
+            replace."package.json" = {
+              description = "package.json overridden";
+              fileMatch = ["package.json"];
+              name = "package.json";
+              url = "https://example.com/package.json";
+            };
+            extra = [
+              {
+                description = "My custom JSON schema";
+                fileMatch = "foo.json";
+                name = "foo.json";
+                url = "https://example.com/schema/foo.json";
+              }
+              {
+                description = "My other custom JSON schema";
+                fileMatch = ["bar.json" ".baar.json"];
+                name = "bar.json";
+                url = "https://example.com/schema/bar.json";
+              }
+            ];
+          };
+        };
+      };
+
+      yaml = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = "Whether to enable the yaml schemas in yamlls.";
+        };
+
+        settings = helpers.mkSettingsOption {
+          options = schemaOpts;
+          description = "Options supplied to the `require('schemastore').yaml.schemas` function.";
+        };
+      };
+    };
+
+    extraConfig = cfg: {
+      plugins.lsp.servers = {
+        # jsonls.settings = mkIf cfg.json.enable {
+        #   schemas.__raw = ''
+        #     require('schemastore').json.schemas({${helpers.toLuaObject cfg.json.settings}})";
+        #   '';
+
+        #   # The plugin recommends to enable this option in its README.
+        #   # Learn more: https://github.com/b0o/SchemaStore.nvim/issues/8
+        #   validate = mkDefault true;
+        # };
+
+        # yamlls.settings = mkIf cfg.yaml.enable {
+        #   schemaStore = {
+        #     # From the README: "You must disable built-in schemaStore support if you want to use
+        #     # this plugin and its advanced options like `ignore`."
+        #     enable = mkDefault false;
+
+        #     # From the README: "Avoid TypeError: Cannot read properties of undefined (reading 'length')"
+        #     url = mkDefault "";
+        #   };
+
+        #   schemas.__raw = ''
+        #     require('schemastore').yaml.schemas({${helpers.toLuaObject cfg.yaml.settings}})";
+        #   '';
+        # };
```

Comment:
Why is this commented ?

Reply:
Without this being uncommented, this module has no real effect on `init.lua`